pkg/controller/v1alpha2/experiment/validator/validator.go

```diff
@@ -1,50 +1,53 @@
-/*
-Copyright 2019 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package util
+package validator
 
 import (
 	"bytes"
 	"database/sql"
 	"fmt"
-	logger "log"
 
 	batchv1beta "k8s.io/api/batch/v1beta1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 
 	commonapiv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/common/v1alpha2"
-	experimentsv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
 	commonv1alpha2 "github.com/kubeflow/katib/pkg/common/v1alpha2"
+	experimentsv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
+	"github.com/kubeflow/katib/pkg/controller/v1alpha2/experiment/manifest"
+	"github.com/kubeflow/katib/pkg/controller/v1alpha2/experiment/util"
 )
 
-func ValidateExperiment(instance *experimentsv1alpha2.Experiment) error {
+var log = logf.Log.WithName("experiment-controller")
+
+type Validator interface {
+	ValidateExperiment(instance *experimentsv1alpha2.Experiment) error
+}
+
+type General struct {
```

Comment:
`General` looks too general to me.  Better name?

Reply:
How about DefaultValidator

Reply:
Is this struct specific to experiments?

Reply:
The interface is only used in validator. And should not be used by other compnents.